src/query/api/v1/middleware/metrics.go

```diff
@@ -91,7 +105,11 @@ func newRouteMetrics(instrumentOpts instrument.Options) *routeMetrics {
 	}
 }
 
-func (m *routeMetrics) metric(path string, status int) (tally.Counter, tally.Timer) {
+func (m *routeMetrics) metric(
+	path string,
+	status int,
+	tags map[string]string,
+) (tally.Counter, tally.Timer) {
 	key := routeMetricKey{
```

Comment:
I think you'll need to add the tags to the route metric key? Also not sure exactly how we'll do that since tags is dynamic here.

Reply:
We might need to use a genny map here so we can overload the `Equals(...)` part of this and do tags equality part of the lookup.